core_domain.process_model_execution/src/resume_process_serivce.ts

```diff
@@ -159,13 +174,62 @@ export class ResumeProcessService implements IResumeProcessService {
     });
   }
 
+  public async retryProcessInstance(identity: IIdentity, processInstanceId: string): Promise<void> {
+    logger.info(`Attempting to retry ProcessInstance with instance ID ${processInstanceId}`);
+
+    const results = await this.processInstanceRepository.query(identity, {
+      processInstanceId: processInstanceId,
+    });
+
+    if (results.length === 0) {
+      throw new NotFoundError(`ProcessInstance with ID ${processInstanceId} not found.`);
+    }
+
+    const processInstance = results[0];
+
+    if (processInstance.state !== ProcessInstanceState.error) {
+      const errorMessage = `ProcessInstance with instance ID ${processInstanceId} can
+       not get restarted: State '${processInstance.state} is not 'error'`;
+      const error = new InternalServerError(errorMessage);
+      logger.error(errorMessage, error);
+      throw error;
+    }
+
+    const flowNodeInstancesForProcessInstance = await this.flowNodeInstanceRepository.query(identity, {
+      processInstanceId: processInstanceId,
+      state: FlowNodeInstanceState.error,
+    });
```

Comment:
Ich würde hier direkt alle FlowNodeInstanzen abfragen und später ein Subset erstellen, in denen die FlowNodeInstanzen mit einem "error" state sind. 

Du wirst die eh alle brauchen, um später nachvollziehen zu können, welche Errors ggf. durch ein ErrorBoundaryEvent abgefangen wurden.

```suggestion
    const flowNodeInstancesForProcessInstance = await this.flowNodeInstanceRepository.query(identity, {
      processInstanceId: processInstanceId,
    });
    
    const erroredFlowNodeInstances = flowNodeInstancesForProcessInstance.filter((entry) => entry.state === FlowNodeInstanceState.error);
```